### PR TITLE
[Merged by Bors] - fix: Operator errors out when credentialsSecret is missing

### PR DESCRIPTION
**Pull request by Maleware**
# Description

Fixing operator errors out if cerdentials Secret not found

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by sbernauer — APPROVED** [review]
LGTM

File: rust/operator-binary/src/superset_db_controller.rs [hunk 1]
```diff
@@ -143,90 +144,74 @@ pub async fn reconcile_superset_db(superset_db: Arc<SupersetDB>, ctx: Arc<Ctx>)
...
-                    .get_opt::<Secret>(
-                        &superset_db.spec.credentials_secret,
-                        superset_db
-                            .namespace()
-                            .as_deref()
-                            .context(ObjectHasNoNamespaceSnafu)?,
-                    )
+                // This is easier to use than `get_opt` and having an Error variant for "Secret does not exist"
+                let _secret = client
+                    .get_opt::<Secret>(&superset_db.spec.credentials_secret, &namespace)
```
Comment (maltesander) [inline rust/operator-binary/src/superset_db_controller.rs:149]:
```suggestion
                    .get::<Secret>(&superset_db.spec.credentials_secret, &namespace)
```

**Review by maltesander — CHANGES_REQUESTED** [review]

**Review by sbernauer — APPROVED** [review]

**Review by maltesander — APPROVED** [review]
Thanks! LGTM!

**Pull request closed without merging**